voter/src/main/java/org/terracotta/voter/ActiveVoter.java

```diff
@@ -170,7 +170,15 @@ ClientVoterManager registerWithActive(String id,
             if (response >= 0) {
               registrationLatch.complete(voterManager);
             } else {
-              LOGGER.warn("Registration with {} in state {} failed. Retrying...", voterManager.getTargetHostPort(), voterManager.getServerState());
+              StringBuilder message = new StringBuilder();
+              message.append(String.format("Registration with %s in state %s failed. ",
+                  voterManager.getTargetHostPort(), voterManager.getServerState()));
+              long voterLimit = Math.max(0, voterManager.getRegisteredVoterLimit());
```

Comment:
When failover priority is set to 'availability', getRegisteredVoterLimit() returns -1.  So I just use the max of '0' to generate the message that "Configured voter limit (0) has been reached"..  Alternatively, I could have prepared a dedicated message just for the -1 case... something like, "voters are not supported when failover priority is set to availability".